spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -489,6 +520,53 @@ class Analyzer(tableUtils: TableUtils,
     }
   }
 
+  // For groupBys validate if the timestamp provided produces some values
+  // if all values are null this should be flagged as an error
+  def runTimestampChecks(df: DataFrame, sampleNumber: Int = 1000): Map[String, String] = {
+
+    val hasTimestamp = df.schema.fieldNames.contains(Constants.TimeColumn)
+    val mapTimestampChecks = if (hasTimestamp) {
+      // set max sample to 1000 rows if larger input is provided
+      val sampleN = if (sampleNumber > 1000) { 1000 }
+      else { sampleNumber }
+      dataFrameToMap(
+        df.limit(sampleN)
+          .agg(
+            // will return 0 if all values are null
+            sum(when(col(Constants.TimeColumn).isNull, lit(0)).otherwise(lit(1)))
+              .cast(StringType)
+              .as("notNullCount"),
+            // assumes that we have valid unix milliseconds between the date range of
+            // 1970-01-01 00:00:00 (0L) to 2099-12-31 23:59:59 (4102473599999L)
+            // will return 0 if all values are within the range
+            sum(when(col(Constants.TimeColumn).between(0L, 4102473599999L), lit(0)).otherwise(lit(1)))
```

Comment:
the current check wouldn't be able to detect milliseconds => seconds mistakes, because all seconds are also valid milliseconds epoch from ~1970s. But this is a very typical mistakes from users.

So I think in practice it might be better to add a lower bound cutoff at 31536000000L (which is milliseconds of 1971-01-01 00:00:00 UTC)

Reply:
done.

Reply:
good catch Haozhen.